core/src/banking_stage/transaction_scheduler/greedy_scheduler.rs

```diff
@@ -96,14 +99,18 @@ impl<Tx: TransactionWithMeta> Scheduler<Tx> for GreedyScheduler<Tx> {
             });
         }
 
+        debug_assert!(
+            self.common.batches.is_empty(),
+            "batches must be empty after scheduling"
```

Comment:
nit: "batches must start empty for scheduling".
weird to say after scheduling when this check is before?

Reply:
heh, I initially put the check _after_ scheduling, but then saw the early returns and moved it. Fixed!